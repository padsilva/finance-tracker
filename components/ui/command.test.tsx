import { render } from "@testing-library/react";

import {
  CommandDialog,
  CommandInput,
  CommandList,
  CommandEmpty,
  CommandGroup,
  CommandItem,
  CommandShortcut,
  CommandSeparator,
} from "./command";

jest.mock("cmdk", () => {
  const Root: React.FC<{ children?: React.ReactNode; className?: string }> = ({
    children,
    className,
  }) => (
    <div data-testid="command-root" className={className}>
      {children}
    </div>
  );
  Root.displayName = "Command";

  const Input: React.FC<
    { className?: string } & React.InputHTMLAttributes<HTMLInputElement>
  > = ({ className, ...props }) => (
    <input data-testid="command-input" className={className} {...props} />
  );
  Input.displayName = "CommandInput";

  const List: React.FC<{ children?: React.ReactNode; className?: string }> = ({
    children,
    className,
  }) => (
    <div data-testid="command-list" className={className}>
      {children}
    </div>
  );
  List.displayName = "CommandList";

  // Return the mock with the same structure as the actual module
  return {
    Command: {
      Root,
      Input: {
        Input,
        displayName: "CommandInput",
      },
      List,
      Empty: ({ children }: { children?: React.ReactNode }) => (
        <div>{children}</div>
      ),
      Group: ({ children }: { children?: React.ReactNode }) => (
        <div>{children}</div>
      ),
      Item: ({ children }: { children?: React.ReactNode }) => (
        <div>{children}</div>
      ),
      Separator: () => <hr />,
      displayName: "Command",
    },
  };
});

describe("Command Components", () => {
  it("should render CommandDialog correctly", () => {
    const { container } = render(
      <CommandDialog>
        <CommandInput placeholder="Search commands..." />
        <CommandList>
          <CommandEmpty>No results found.</CommandEmpty>
        </CommandList>
      </CommandDialog>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render CommandGroup with custom styling", () => {
    const { container } = render(
      <CommandGroup className="custom-group" heading="Custom Group">
        <CommandItem>Item 1</CommandItem>
        <CommandItem>Item 2</CommandItem>
      </CommandGroup>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render CommandItem in different states", () => {
    const { container } = render(
      <>
        <CommandItem>Default Item</CommandItem>
        <CommandItem data-selected={true}>Selected Item</CommandItem>
        <CommandItem data-disabled={true}>Disabled Item</CommandItem>
        <CommandItem className="custom-item">Custom Item</CommandItem>
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render CommandShortcut with custom className", () => {
    const { container } = render(
      <CommandShortcut className="custom-shortcut">âŒ˜K</CommandShortcut>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render CommandSeparator with custom styling", () => {
    const { container } = render(
      <CommandSeparator className="custom-separator" />,
    );

    expect(container).toMatchSnapshot();
  });
});
