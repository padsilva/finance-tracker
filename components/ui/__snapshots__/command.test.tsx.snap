// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command Components should render CommandDialog correctly 1`] = `<div />`;

exports[`Command Components should render CommandGroup with custom styling 1`] = `
<div>
  <div>
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
  </div>
</div>
`;

exports[`Command Components should render CommandItem in different states 1`] = `
<div>
  <div>
    Default Item
  </div>
  <div>
    Selected Item
  </div>
  <div>
    Disabled Item
  </div>
  <div>
    Custom Item
  </div>
</div>
`;

exports[`Command Components should render CommandSeparator with custom styling 1`] = `
<div>
  <hr />
</div>
`;

exports[`Command Components should render CommandShortcut with custom className 1`] = `
<div>
  <span
    class="ml-auto text-xs tracking-widest text-muted-foreground custom-shortcut"
  >
    âŒ˜K
  </span>
</div>
`;
